### PR TITLE
Kinetic Accelerators Auto-Reload

### PR DESCRIPTION
**Pull request by Fox-McCloud**
port of: https://github.com/tgstation/-tg-station/pull/15213

Kinetic Accelerators (and by virtue of inheritance, e-bows) automatically reload instead of having to be manually clicked and reloaded.

:cl: Fox McCloud
Tweak: Kinetic Accelerators and E-bows automatically reload
/:cl:

### CODE REVIEW

**Comment by Spacemanspark** [comment]
This won't give me conflict issues, will it?

**Comment by Crazylemon64** [comment]
Oh, that's nifty - might give me a little less reason to play with hotkeys all the time

File: code/modules/projectiles/guns/energy/special.dm [hunk 1]
```diff
@@ -229,26 +225,23 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
 	flags = NODROP
 
 /obj/item/weapon/gun/energy/kinetic_accelerator/Fire()
-	overheat = 1
-	spawn(overheat_time)
-		overheat = 0
-		recent_reload = 0
+	if(!recently_fired)
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/energy/special.dm]: For those of you wondering why this is here, simply put, our gun code is shit, so if you can somehow get `Fire` to be called, again, then you would be able to double up on the the amount of reloads you could get (this is what caused the uber fast reload that @TheDZD had to correct for, originally).

**Comment by Crazylemon64** [comment]
@Spacemanspark This touches guncode only, your PR should be fine.

**Comment by Spacemanspark** [comment]
Ya, I just got too used to looking at /vg/'s code page for the mining items.

**Comment by Fox-McCloud** [comment]
mini-rant about gun code: ideally, you'd have the spawn(fire_delay) reload() take place AFTER you fire the shot, if you want to be logically consistent about it.

Problem: something stupid is handled with a sleep; probably fire_delay itself, thus resulting in obscenely long recharge times if put the self recharging after the parent call.

Our gun code is so horrible.

File: code/modules/projectiles/guns/energy/special.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/energy/special.dm]: mini-rant about gun code: ideally, you'd have the spawn(fire_delay) reload() take place AFTER you fire the shot, if you want to be logically consistent about it.

Problem: something stupid is handled with a sleep; probably fire_delay itself, thus resulting in obscenely long recharge times if put the self recharging after the parent call.

Our gun code is so horrible from a back-end standpoint...

**Pull request merged**